scripts/list_kbase_users.py

```diff
@@ -0,0 +1,206 @@
+    users = set(ret['result'][0]['users'])
+
+    ujobs = set()
+    count = 0
+    for j in db.jobstate.find():
+        ujobs.add(j['user'])
+        cb = j.get('canceledby')
+        if cb:
+            ujobs.add(cb)
+        sh = j.get('shared')
+        if sh:
+            ujobs.update(sh)
+        count += 1
+        if count % 100000 == 0:
+            print 'At job record ' + str(count)
+    users.update(ujobs)
+    return users
+
+
+def proc_shock(cfg):
+    db = get_mongo_db(cfg, 'shock', 'mongodb-host', 'mongodb-database', 'mongodb-user',
+                      'mongodb-pwd')
+
+    ret = db.Users.aggregate([{'$group': {'_id': None, 'users': {'$push': '$username'}}}])
+    users = set(ret['result'][0]['users'])
+    users.remove('*')
+    users.remove('')
+    users.remove('1')
+    return users
+
+
+def proc_awe(cfg):
+    db = get_mongo_db(cfg, 'awe', 'mongodb-host', 'mongodb-database', 'mongodb-user',
+                      'mongodb-pwd')
+
+    ret = db.Users.aggregate([{'$group': {'_id': None, 'users': {'$push': '$username'}}}])
+    users = set(ret['result'][0]['users'])
+    print 'awe'
+    print sorted(users)
+    return users
+
+
+def proc_userprof(cfg):
+    db = get_mongo_db(cfg, 'UserProfile', 'mongodb-host', 'mongodb-database', 'mongodb-user',
+                      'mongodb-pwd')
+
+    ret = db.profiles.aggregate([{'$group': {'_id': None, 'users': {'$push': '$user.username'}}}])
+    users = set(ret['result'][0]['users'])
+    return users
+
+
+def proc_catalog_nms(cfg):
+    db = get_mongo_db(cfg, 'catalog', 'mongodb-host', 'mongodb-database', 'mongodb-user',
+                      'mongodb-pwd')
+
+    ret = db.developers.aggregate([{'$group': {'_id': None, 'users': {'$push': '$kb_username'}}}])
+    users = set(ret['result'][0]['users'])
+
+    ret = db.exec_stats_raw.aggregate([{'$group': {'_id': None, 'users': {'$push': '$user_id'}}}])
+    ustatsraw = set(ret['result'][0]['users'])
+    users.update(ustatsraw)
+
+    ret = db.exec_stats_users.aggregate([{'$group': {'_id': None, 'users':
+                                                     {'$push': '$user_id'}}}])
+    ustatsuser = set(ret['result'][0]['users'])
+    users.update(ustatsuser)
+
+    ret = db.favorites.aggregate([{'$group': {'_id': None, 'users': {'$push': '$user'}}}])
+    ufav = set(ret['result'][0]['users'])
+    users.update(ufav)
+
+    authors = set()
+    count = 0
+    for a in db.local_functions.find():
+        authors.update(a['authors'])
+        count += 1
+        if count % 100000 == 0:
+            print 'At local_functions record ' + str(count)
+    users.update(authors)
+
+    modowners = set()
+    count = 0
+    for a in db.modules.find():
+        for o in a['owners']:
+            modowners.add(o['kb_username'])
+        count += 1
+        if count % 100000 == 0:
+            print 'At modules record ' + str(count)
+    users.update(modowners)
+
+    repoowners = set()
+    count = 0
```

Comment:
No, I mean that it may happen that admins separate catalog and nms databases. As I thought even if there is no collection mango treats it as empty collection. Correct me if I'm wrong.